src/CameraComponent_V4L2.cpp

```diff
@@ -0,0 +1,467 @@
+/*
+ * This file is part of the Camera Streaming Daemon
+ *
+ * Copyright (C) 2017  Intel Corporation. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <assert.h>
+#include <fcntl.h>
+#include <linux/videodev2.h>
+#include <sstream>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <unistd.h>
+
+#include "mavlink_server.h"
+
+#include "CameraComponent_V4L2.h"
+#include "log.h"
+
+CameraComponent_V4L2::CameraComponent_V4L2()
+    : CameraComponent()
+{
+}
+
+CameraComponent_V4L2::CameraComponent_V4L2(std::string devicepath)
+    : CameraComponent()
+{
+    log_debug("%s path:%s", __func__, devicepath.c_str());
+}
+
+CameraComponent_V4L2::CameraComponent_V4L2(std::string devicepath, std::string uri)
+    : CameraComponent()
+{
+
+    log_debug("%s path:%s uri:%s", __func__, devicepath.c_str(), uri.c_str());
+    // TODO:: get the supported paramters from the camera device
+    // Initialize the supported and default values
+    dev_path = devicepath;
+    strcpy((char *)camInfo.cam_definition_uri, uri.c_str());
```

Comment:
strcpy are not safe. you can have a buffer overflow here if uri.size() > sizeof(camInfo.cam_definition_uri).